src/helpers/parallel.ml

```diff
@@ -21,6 +28,25 @@ let decode of_yojson data =
   with
   | exn -> Error exn
 
+let protocol_decode_list of_yojson data =
```

Comment:
why not do this with `Task.parallel_for?

Reply:
I don't want to deal with numbers or mutation and parallel_for requires both, we actually had problems regarding that, parallel_for is like a for loop

Reply:
We use libraries that leverage mutability all the time and still going, besides it's not like the code for parallel_map will be modified all the time.
code for parallel_map is taken from domainslib pr: 
```ocaml
let parallel_map arr ~pool ~f=
  let open Task in
  let len = Array.length arr in
  let res = Array.make len (f arr.(0)) in
  Task.parallel_for ~chunk_size ~start:1 ~finish:(len - 1)
  ~body:(fun i ->
    res.(i) <- (f arr.(i))) pool;
  res
 ```
 and then the code becomes(assuming user_operations are represented as `string array` instead of `string list`: 
 ```ocaml
 let protocol_decode_batch data ~of_yojson  = 
   parallel_map data 
   ~pool:(domainslib_pool ()) 
   ~f:(fun elem ->
               match decode of_yojson data with 
               | Ok (Ok data) -> Ok data
               | Ok (Error error) -> Error (`Of_yojson error)
               | Error exn -> Error (`Of_exception exn))
  |> Array.to_list
 ```